internal/testUI/main.go

```diff
@@ -0,0 +1,389 @@
+	ui.mu.Lock()
+	defer ui.mu.Unlock()
+
+	// Build our config and set our options
+	// cfg := &config{Writer: color.Output}
+	// for _, opt := range opts {
+	// 	opt(cfg)
+	// }
+
+	table := tablewriter.NewWriter(ui.OutWriter)
+	table.SetHeader(tbl.Headers)
+	table.SetBorder(false)
+	table.SetAutoWrapText(false)
+
+	for _, row := range tbl.Rows {
+		colors := make([]tablewriter.Colors, len(row))
+		entries := make([]string, len(row))
+
+		for i, ent := range row {
+			entries[i] = ent.Value
+
+			color, ok := colorMapping[ent.Color]
+			if ok {
+				colors[i] = tablewriter.Colors{color}
+			}
+		}
+
+		table.Rich(entries, colors)
+	}
+
+	table.Render()
+}
+
+// Debug implements UI
+func (ui *nonInteractiveTestUI) Debug(msg string) {
+	ui.Output(msg, terminal.WithDebugStyle())
+}
+
+// Error implements UI
+func (ui *nonInteractiveTestUI) Error(msg string) {
+	ui.Output(msg, terminal.WithErrorStyle())
+}
+
+// ErrorWithContext satisfies the ErrorWithContext function on the UI
+// interface.
+func (ui *nonInteractiveTestUI) ErrorWithContext(err error, sub string, ctx ...string) {
+	ui.Error(strings.Title(sub))
+	ui.Error("  Error: " + err.Error())
+	ui.Error("  Context:")
+	max := 0
+	for _, entry := range ctx {
+		if loc := strings.Index(entry, ":") + 1; loc > max {
+			max = loc
+		}
+	}
+	for _, entry := range ctx {
+		padding := max - strings.Index(entry, ":") + 1
+		ui.Error("  " + strings.Repeat(" ", padding) + entry)
+	}
+}
+
+// Header implements UI
+func (ui *nonInteractiveTestUI) Header(msg string) {
+	ui.Output(msg, terminal.WithHeaderStyle())
+}
+
+// Info implements UI
+func (ui *nonInteractiveTestUI) Info(msg string) {
+	ui.Output(msg, terminal.WithInfoStyle())
+}
+
+// Success implements UI
+func (ui *nonInteractiveTestUI) Success(msg string) {
+	ui.Output(msg, terminal.WithSuccessStyle())
+}
+
+// Trace implements UI
+func (ui *nonInteractiveTestUI) Trace(msg string) {
+	ui.Output(msg, terminal.WithTraceStyle())
+}
+
+// Warning implements UI
+func (ui *nonInteractiveTestUI) Warning(msg string) {
+	ui.Output(msg, terminal.WithWarningStyle())
+}
+
+// WarningBold implements UI
+func (ui *nonInteractiveTestUI) WarningBold(msg string) {
+	ui.Output(msg, terminal.WithStyle(terminal.WarningBoldStyle))
+}
+
+type nonInteractiveStatus struct {
+	mu *sync.Mutex
+}
+
+func (s *nonInteractiveStatus) Update(msg string) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	fmt.Fprintln(color.Output, msg)
+}
+
+func (s *nonInteractiveStatus) Step(status, msg string) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	fmt.Fprintf(color.Output, "%s: %s\n", textStatus[status], msg)
+}
+
+func (s *nonInteractiveStatus) Close() error {
+	return nil
+}
+
+type nonInteractiveStepGroup struct {
+	mu     *sync.Mutex
+	wg     sync.WaitGroup
+	closed bool
+}
+
+// Start a step in the output
+func (f *nonInteractiveStepGroup) Add(str string, args ...interface{}) terminal.Step {
+	// Build our step
+	step := &nonInteractiveStep{mu: f.mu}
```

Comment:
This is just for tests, so probably not a big deal, but it's usually better to avoid sharing mutexes. It can lead to unexpected deadlocks.